### PR TITLE
[Microsoft ATP] enhancement - adding query_batch argument to command

### PR DESCRIPTION
**Pull request by MLainer1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3869)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 1]
```diff
@@ -853,26 +853,54 @@ script:
...
       defaultValue: ''
       predefined:
       - ''
-    deprecated: false
+    - default: false
+      description:  |-
+        A JSON array of queries, limited to 10 queries. Must be passed if query argument is empty. Example for input:
+        { "queries": [
+            {
+            "query": "query #1",	// Mandatory
```
Comment (dantavori) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:880]: let's remove the comments here and describe below which field is mandatory or not.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 2]
```diff
@@ -853,26 +853,54 @@ script:
...
+      description: Time range to look back. Expected syntax is a human readable time range, e.g. 60 minutes, 6 hours, 1 day, etc. If specified with query_batch, will be applied to all queries in the array.
       required: false
       isArray: false
       defaultValue: ''
       predefined:
       - ''
-    deprecated: false
+    - default: false
+      description:  |-
+        A JSON array of queries, limited to 10 queries. Must be passed if query argument is empty. Example for input:
```
Comment (dantavori) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:877]:
```suggestion
        A JSON array of queries, limited to 10 queries. Cannot be provided with the query argument. Example for input:
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 3]
```diff
@@ -2547,19 +2548,52 @@ def get_advanced_hunting_command(client: MsClient, args: dict):
...
+            query = json.loads(query_batch).get('queries')
+            queries.extend(query)
+        except Exception as e:
+            raise DemistoException('Please provide a valid JSON query_batch.', e)
+
+    if len(queries) > 10:
+        raise DemistoException('Please provide only up to 10 queries.')
+
+    human_readable = ''
+    outputs = []
```
Comment (dantavori) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:2575]:
```suggestion
    outputs: List[Dict[str, Any]] = []
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 4]
```diff
@@ -2547,19 +2548,52 @@ def get_advanced_hunting_command(client: MsClient, args: dict):
...
+        response = client.get_advanced_hunting(query, timeout, time_range)
+        results = response.get('Results')
+        if isinstance(results, list) and len(results) == 1:
+            report_id = results[0].get('ReportId')
+            if report_id:
+                results[0]['ReportId'] = str(report_id)
+        if name:
+            outputs.append({name: results})
+        else:
+            outputs = results  # type: ignore
```
Comment (dantavori) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:2591]:
```suggestion
            outputs = results
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 5]
```diff
@@ -853,26 +853,54 @@ script:
...
+            }
+          ]
+        }
+      isArray: false
+      name: query_batch
+      required: false
+      secret: false
+      deprecated: false
+    - default: false
+      description: If stated along with query, then the response should be saved in context under the Result.name path.
```
Comment (dantavori) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:899]: let's think how to rephrase this so the user will understand how it's stored

**Review by dantavori — CHANGES_REQUESTED** [review]

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py (hunk 4 above)
Comment (MLainer1) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:2590]: Doesn’t really work

**Comment by ShirleyDenkberg** [comment]
@dantavori Doc review completed.

**Review by dantavori — APPROVED** [review]

**Pull request merged**